assets/js/blocks/checkout/inner-blocks/checkout-payment-block/edit.tsx

```diff
@@ -38,9 +35,18 @@ export const Edit = ( {
 	};
 	setAttributes: ( attributes: Record< string, unknown > ) => void;
 } ): JSX.Element => {
-	const globalPaymentMethods = getSetting(
+	const globalPaymentMethods = getSetting< GlobalPaymentMethod[] >(
 		'globalPaymentMethods'
-	) as paymentAdminLink[];
+	);
+
+	const { incompatiblePaymentMethods } = useSelect( ( select ) => {
+		const { getIncompatiblePaymentMethods } = select( PAYMENT_STORE_KEY );
+		return {
+			incompatiblePaymentMethods: getIncompatiblePaymentMethods(),
+		};
+	}, [] );
+	const incompatiblePaymentMethodMessage =
+		'Incompatible with block-based checkout';
```

Comment:
```suggestion
		__( 'Incompatible with block-based checkout', 'woo-gutenberg-products-block' );
```

I blieve that all messages should be translatable by default.